From: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Date: Fri, 18 Nov 2022 14:31:36 +0800
Subject: perf/x86/intel/uncore: Fix reference count leak in
 snr_uncore_mmio_map()
Git-commit: 8ebd16c11c346751b3944d708e6c181ed4746c39
Patch-mainline: v6.2-rc1
References: jsc#PED-5023 bsc#1211439 (git-fixes)
X-Info: adjust for context, no "Factor out snr_uncore_mmio_map" 1583971b5cb8c786df88be580cdd96a974ad591b; no "Record the size of mapped area" 1b94d31de422399421422af0e63c9685e7485901

pci_get_device() will increase the reference count for the returned
pci_dev, so snr_uncore_get_mc_dev() will return a pci_dev with its
reference count increased. We need to call pci_dev_put() to decrease the
reference count. Let's add the missing pci_dev_put().

Fixes: ee49532b38dd ("perf/x86/intel/uncore: Add IMC uncore support for Snow Ridge")
Signed-off-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Signed-off-by: Peter Zijlstra (Intel) <peterz@infradead.org>
Reviewed-by: Kan Liang <kan.liang@linux.intel.com>
Link: https://lore.kernel.org/r/20221118063137.121512-4-wangxiongfeng2@huawei.com

Signed-off-by: Tony Jones <tonyj@suse.de>
---
 arch/x86/events/intel/uncore_snbep.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/arch/x86/events/intel/uncore_snbep.c
+++ b/arch/x86/events/intel/uncore_snbep.c
@@ -4424,6 +4424,8 @@
 
 	addr += box_ctl;
 
+	pci_dev_put(pdev);
+
 	box->io_addr = ioremap(addr, SNR_IMC_MMIO_SIZE);
 	if (!box->io_addr)
 		return;
